pySDC/tests/test_projects/test_pintsime/test_SwitchEstimator.py

```diff
@@ -0,0 +1,464 @@
+        dt=dt,
+        problem=problem,
+        sweeper=sweeper,
+        num_nodes=num_nodes,
+        quad_type=quad_type,
+        QI=QI,
+        hook_class=hook_class,
+        use_adaptivity=useA,
+        use_switch_estimator=useSE,
+        problem_params=problem_params,
+        restol=restol,
+        maxiter=maxiter,
+        max_restarts=max_restarts,
+        tol_event=tol,
+        alpha=alpha,
+    )
+
+    stats, t_switch_exact = controllerRun(
+        description=description,
+        controller_params=controller_params,
+        controller=controller,
+        t0=t0,
+        Tend=Tend,
+        exact_event_time_avail=exact_event_time_avail,
+    )
+
+    # in this specific example only one event has to be found
+    switches = [me[1] for me in get_sorted(stats, type='switch', sortby='time', recomputed=False)]
+    assert len(switches) >= 1, f'{problem.__name__}: No events found for tol={tol}!'
+
+    t_switch = switches[-1]
+    event_err = abs(t_switch - t_switch_exact)
+    assert np.isclose(event_err, 0, atol=1.2e-11), f'Event time error {event_err} is not small enough!'
+
+
+@pytest.mark.base
+@pytest.mark.parametrize('num_nodes', [3, 4, 5])
+def testDetectionDAE(num_nodes):
+    r"""
+    In this test, the switch estimator is applied to a DAE dummy problem of ``DiscontinuousTestDAE``,
+    where the dynamics of the differential equation is replaced by its exact dynamics to see if
+    the switch estimator predicts the event correctly. The problem is tested for a combination
+    of different tolerances ``tol`` and different number of collocation nodes ``num_nodes``.
+
+    Since the problem only uses the exact dynamics, the event should be predicted very accurately
+    by the switch estimator.
+
+    Parameters
+    ----------
+    tol : float
+        Tolerance for switch estimator.
+    num_nodes : int
+        Number of collocation nodes.
+    quad_type : str
+        Type of quadrature.
+    """
+
+    from pySDC.projects.DAE.sweepers.fully_implicit_DAE import fully_implicit_DAE
+    from pySDC.projects.DAE.problems.DiscontinuousTestDAE import DiscontinuousTestDAE
+    from pySDC.helpers.stats_helper import get_sorted
+    from pySDC.projects.PinTSimE.battery_model import generateDescription, controllerRun
+    from pySDC.implementations.hooks.log_solution import LogSolution
+    from pySDC.implementations.hooks.log_restarts import LogRestarts
+    from pySDC.projects.DAE.misc.HookClass_DAE import LogGlobalErrorPostStepDifferentialVariable
+    from pySDC.projects.PinTSimE.paper_PSCC2024.log_event import LogEventDiscontinuousTestDAE
+
+    problem = DiscontinuousTestDAE
+    problem_params = dict({'newton_tol': 1e-6})
+    t0 = 4.6
+    Tend = 4.62
+    dt = Tend - t0
+    tol = 1e-10
+
+    sweeper = fully_implicit_DAE
+    QI = 'LU'
+    quad_type = 'RADAU-RIGHT'
+
+    _, _, _, _, useA, useSE, exact_event_time_avail = getParamsRun()
+
+    restol = 1e-13
```

Comment:
A residual tolerance this low will probably enforce maximum iterations on some architectures. Maybe this test could be a bit faster with a slightly larger tolerance or a lower number of maximum iterations?

Reply:
Hm. Ah, yes. Since the state function here only depends on the differential variable, we could use a larger residual tolerance to speed up the test a bit. Thanks for the hint!